clang/lib/Sema/SemaFunctionEffects.cpp

```diff
@@ -1259,6 +1263,12 @@ class Analyzer {
       return true;
     }
 
+    bool TraverseStmt(Stmt *Statement) {
+      if (Statement != TrailingRequiresClause)
```

Comment:
Could consider leaving a comment here saying why you skip trailing requires clauses (some truncated version of the description/findings of this PR)